lib/sdk.js

```diff
@@ -308,23 +315,28 @@ Vantiq.prototype.subscribe = function(resource, name, operation, parameters, cal
             parameters = {};
         } else if (callback === undefined) {
             callback = parameters;
-            parameters = operation;
-            operation = null;
-            parameters = {};
+            if (typeof operation === "string") {
+                return Promise.reject(new Error('Operation only supported for "types"'));
+            } else {
+                parameters = operation;
+                operation = null;
+            }
+        } else if (callback !== undefined && parameters !== undefined && operation !==undefined) {
+            return Promise.reject(new Error('Operation only supported for "types"'));
         }
         break;
     case 'types':
         if(operation === undefined || !operation instanceof  String) {
-            return Promise.reject(new Error('Operation required for "types"'));
+            return Promise.reject(new Error('Operation must be "insert", "update" or "delete"'));
```

Comment:
why was this error text changed?  And the error text below?
They seem to have been switched, but I think they make more sense using the original text.

Reply:
I think they got switched at some point and I switched it back. One of the Travis tests expected these particular errors messages where they are

Reply:
The original code from 2 versions ago (b1fa060) was:
```
        if(arguments.length != 4) {
            return Promise.reject(new Error('Operation required for "types"'));
        }
        if(operation !== 'insert' &&
           operation !== 'update' &&
           operation !== 'delete') {
            return Promise.reject(new Error('Operation must be "insert", "update" or "delete"'));            
        }
```
So I think the strings should be put back to what they were then...
Since you added a fifth argument, maybe the first check should just be:
```
        if(arguments.length < 4) {
            return Promise.reject(new Error('Operation required for "types"'));
        }
```
and then the later check can go back to:
```
        if(operation !== 'insert' &&
           operation !== 'update' &&
           operation !== 'delete') {
            return Promise.reject(new Error('Operation must be "insert", "update" or "delete"'));            
        }
```
and that should get the tests to pass.

It would also be good to add a test-case that gets that 2nd error, to make sure that it doesn't get lost again.
Something like:
```
        it('can ensure subscribe on types have valid operations', function() {
            return p.then(function() {
                return v.subscribe('types', 'foo', 'dummyOp2', () => {})
                    .catch((err) => {
                        err.message.should.equal('Operation must be "insert", "update" or "delete"');
                    });
            });
        });

```